include/boost/system/detail/error_code.hpp

```diff
@@ -114,7 +114,10 @@ class error_code
 
     // constructors:
 
-    BOOST_SYSTEM_CONSTEXPR error_code() BOOST_NOEXCEPT:
+#if ! defined(BOOST_GCC) || BOOST_GCC >= 40800
```

Comment:
BOOST_WORKAROUND would probably be better here.

Reply:
Yep, Emil suggested that too but I erred on the side of not adding an additional include. I'm happy to use that macro though.

Reply:
Please do.

Reply:
oh dang, system/detail/config.hpp already includes workaround.hpp.

Reply:
Doesn't matter, you should include it here too because it's used.